### ISSUE TITLE
Randoop captures OutOfMemoryError for regression testing

### ISSUE THREAD

**Issue by rjust** [randoop/randoop#72]
If Randoop runs out of memory during test generation, it catches the `OutOfMemoryError` and generates a test that expects it. For example:
```
@Test                                                                          
  public void test314() throws Throwable {                                       
    if (debug) { System.out.format("%n%s%n","RandoopTest2.test314"); }                                                                      
    // The following exception was thrown during execution.                      
    // This behavior will recorded for regression testing.                       
    try {                                                                        
      org.apache.commons.math3.linear.Array2DRowRealMatrix array2DRowRealMatrix2 = 
         new org.apache.commons.math3.linear.Array2DRowRealMatrix(1755678109, 2147483647);
      org.junit.Assert.fail("Expected exception of type java.lang.OutOfMemoryError");
    } catch (java.lang.OutOfMemoryError e) {                                     
      // Expected exception.                                                     
    }                                                                                                                                                          
  }
```

This is problematic for two reasons:
1) A test might not trigger an OutOfMemoryError if executed after generation.
2) Continuing test generation and capturing the system state after an OutOfMemoryError occurred seems unreliable.

**Comment by bjkeller** [comment]
Thanks René.  Do you have any suggestions on desired behavior when this exception is encountered?

**Comment by mernst** [comment]
In this particular case, catching the OutofMemoryError seems right -- that matrix is so big it will always trigger the exception.  Do you have an example where this behavior is problematic?

I can see multiple desires for treatment of OutOfMemoryError:
  * On the one hand, OutOfMemoryError is like any other exception and should be treated as such.  The user can use the `--regression-assertions-about-exceptions` command-line option to affect treatment of exceptions.  (We could customize it to enable specific treatment of particular exceptions.)
  * On the other hand, Randoop may be more likely to throw such an exception during generation than the test suite is likely to throw the exception during execution.  That's mostly true if Randoop remembers all the values in the pool.  But I thought that it only remembered the sequences, not the Java values.  (The size of the sequences in memory could still be large, though.)

**Comment by bjkeller** [comment]
Mike, 
On your last point, each test sequence is (currently) represented as an ExecutableSequence, which has the values for the last execution of the sequence.  So, the generated tests might be hauling around values that aren't going to be used again since the checks are already generated.  This is on my desired refactor list, but for other reasons.

**Comment by rjust** [comment]
The following test did trigger an `OutOfMemoryError` during test generation, but it doesn't during any later execution. Instead, it doesn't terminate (at least within 30min).
```
@Test
public void test1() throws Throwable {                                         
    if (debug) { System.out.println(); System.out.print("RandoopTest2265.test1"); }                                                                
    // The following exception was thrown during execution.                      
    // This behavior will recorded for regression testing.                       
    try {                                                                        
      double var2 = org.apache.commons.math3.util.Precision.round(1201.0d, 171474276);
      fail("Expected exception of type java.lang.OutOfMemoryError");             
    } catch (java.lang.OutOfMemoryError e) {                                     
      // Expected exception.                                                     
    }                                                                                                                                                         
  } 
```

**Comment by bjkeller** [comment]
Fixed in v2.1.0. Use option --oom-exception to indicate whether you want an OutOfMemoryError to be INVALID (test is discarded), ERROR (error-revealing test), or EXPECTED (regression test)
